config/hardware/probes/beacon_contact.cfg

```diff
@@ -0,0 +1,59 @@
+# This probe type is for a Beacon probe used directly as a virtual Z endstop
+# rather than with an existing physical endstop. To use this configuration,
+# you will need to manually add the Beacon Klipper plugin!
+
+## Then, you should just add the following two lines to your overrides and everything should work!
+## The rest of the allowed config entries are available on this link: config/hardware/probes/inductive_virtual.cfg
+# [beacon]
+# serial: /dev/serial/by-id/usb-Beacon_Beacon_...
+
+
+[gcode_macro _USER_VARIABLES]
+# We can declare an "inductive_virtual" probe type as it's pretty close to the Beacon way of working and should just work!
+variable_probe_type_enabled: "beacon_contact"
+variable_startprint_actions: "extruder_preheating", "bed_soak", "chamber_soak", "tilt_calib", "bedmesh", "extruder_heating", "purge", "clean", "primeline"
+gcode:
+
+# Beacon probe definition also include the probe management macros directly from here
+[include ../../../macros/base/probing/beacon_probe.cfg]
+[include ../../../macros/base/probing/overrides/beacon_probe_overrides.cfg]
+[include ../../../macros/base/homing/beacon_homing_hooks.cfg]
+
+[stepper_z]
+endstop_pin: probe:z_virtual_endstop
+homing_retract_dist: 0
+
+[beacon]
+home_xy_position: 117.5, 117.5   # update with your safe position
+home_z_hop: 5
+home_z_hop_speed: 30
+home_xy_move_speed: 300
+home_y_before_x: false
+home_method: contact
+home_method_when_homed: proximity
+home_autocalibrate: unhomed
+home_gcode_pre_x: _HOME_PRE_AXIS AXIS=X
+home_gcode_post_x: _HOME_POST_AXIS AXIS=X
+home_gcode_pre_y: _HOME_PRE_AXIS AXIS=Y
+home_gcode_post_y: _HOME_POST_AXIS AXIS=Y
+home_gcode_pre_xy: _HOME_XY STAGE=pre
+home_gcode_post_xy: _HOME_XY STAGE=post
+contact_activate_gcode: _CONTACT_MACRO STAGE=activate
+contact_deactivate_gcode: _CONTACT_MACRO STAGE=deactivate
+
+[gcode_macro _beacon_contact_hardware_check]
+gcode:
+    {% if 'mcu beacon' in printer %}
+        {% set version = printer['mcu beacon'].mcu_version.split(' ')|last %}
+        {% set major = version.split('.')|first|int %}
+        {% if version < '2' %}
+            {action_raise_error('Beacon firmware is too old. Found: %s, Expected: 2.0.0 or higher' % version)}
+        {% endif %}
+    {% else %}
+        {action_raise_error('Beacon MCU was not found')}
+    {% endif %}
+
+[delayed_gcode _do_beacon_contact_hardware_check]
+initial_duration: 5
+gcode:
+    _beacon_contact_hardware_check
```

Comment:
Is it possible to move this check in the startup.cfg file? There's already a few things that are checked at Klippain startup and this could be added in the _INIT_CHECKPROBECONF macro that is done exactly for this purpose.

https://github.com/Frix-x/klippain/blob/main/macros/miscs/startup.cfg